packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -510,9 +527,12 @@ abstract class VpcBase extends Resource implements IVpc {
    * PUBLIC (in that order) that has any subnets.
    */
   private reifySelectionDefaults(placement: SubnetSelection): SubnetSelection {
+
     if (placement.subnetName !== undefined) {
       if (placement.subnetGroupName !== undefined) {
         throw new Error('Please use only \'subnetGroupName\' (\'subnetName\' is deprecated and has the same behavior)');
+      } else {
+        Annotations.of(this).addWarning('Usage of \'subnetName\' in SubnetSelection is deprecated, use \'subnetGroupName\' instead');
```

Comment:
This is API contract misuse and should definitely remain an error that aborts the program.

Reply:
The error still occurs if the user specifies both, like before. Now the user will get a friendly reminder that `subnetName` shouldn't be used, but still allows them to use it -- like before.

Should I change the behavior such that it throws an error?

Reply:
Oh I see, I misread.

Well, hmm. I guess it's maybe a little overkill, but fine.